.github/workflows/cpu_test.yml

```diff
@@ -0,0 +1,83 @@
+name: CPU Tests
+
+on:
+  # Trigger the workflow on push to master or any pull request
+  push:
+    branches:
+      - master
```

Comment:
What do you think about triggering it for all branches? Some of the branches aren't master but are important for the project. Also, this will help to understand CI failures before opening up the PR itself.

Reply:
Yeah we can do that.